packages/backend/src/accounting/psql/ledger-transfer/index.test.ts

```diff
@@ -49,6 +69,322 @@ describe('Ledger Transfer', (): void => {
+      }
+
+      transfer.creditAccount.id = ''
+
+      await expect(
+        createTransfers(serviceDeps, [transfer], knex)
+      ).rejects.toThrow('Could not create transfer(s)')
+    })
+
+    test.each`
+      amount | description
+      ${0n}  | ${'zero'}
+      ${-1n} | ${'negative'}
+    `(
+      'returns error for invalid amount ($description)',
+      async ({ amount }: { amount: bigint }): Promise<void> => {
+        const transfer: CreateTransferArgs = {
+          ...baseTransfer,
+          amount
+        }
+
+        await expect(
+          createTransfers(serviceDeps, [transfer], knex)
+        ).resolves.toEqual({
+          results: [],
+          errors: [{ index: 0, error: TransferError.InvalidAmount }]
+        })
+      }
+    )
+
+    test('returns error for mismatched account assets', async (): Promise<void> => {
+      const transfer: CreateTransferArgs = {
+        ...baseTransfer
+      }
+
+      transfer.creditAccount['ledger' as string] = 1
+      transfer.debitAccount['ledger' as string] = 0
+
+      await expect(
+        createTransfers(serviceDeps, [transfer], knex)
+      ).resolves.toEqual({
+        results: [],
+        errors: [{ index: 0, error: TransferError.DifferentAssets }]
+      })
+    })
+
+    test('returns error if same account transfer', async (): Promise<void> => {
+      const transfer: CreateTransferArgs = {
+        ...baseTransfer,
+        creditAccount: account,
+        debitAccount: account
+      }
+
+      await expect(
+        createTransfers(serviceDeps, [transfer], knex)
+      ).resolves.toEqual({
+        results: [],
+        errors: [{ index: 0, error: TransferError.SameAccounts }]
+      })
+    })
+
+    test('returns error for negative timeout value', async (): Promise<void> => {
+      const transfer: CreateTransferArgs = {
+        ...baseTransfer,
+        timeoutMs: -1n
+      }
+
+      await expect(
+        createTransfers(serviceDeps, [transfer], knex)
+      ).resolves.toEqual({
+        results: [],
+        errors: [{ index: 0, error: TransferError.InvalidTimeout }]
+      })
+    })
+
+    test('does not create transfers if any fail', async (): Promise<void> => {
+      const transfer: CreateTransferArgs = {
+        ...baseTransfer
+      }
+
+      const failTransfer: CreateTransferArgs = {
+        ...baseTransfer,
+        transferRef: uuid(),
+        creditAccount: account,
+        debitAccount: account
+      }
+
+      await expect(
+        createTransfers(serviceDeps, [transfer, failTransfer], knex)
+      ).resolves.toEqual({
+        results: [],
+        errors: [{ index: 1, error: TransferError.SameAccounts }]
+      })
+
+      await expect(
+        LedgerTransfer.query(knex).findOne({
+          transferRef: failTransfer.transferRef
+        })
+      ).resolves.toBeUndefined()
+
+      await expect(
+        LedgerTransfer.query(knex).findOne({
+          transferRef: transfer.transferRef
+        })
+      ).resolves.toBeUndefined()
```

Comment:
Alternatively
```suggestion
      await expect(
        getAccountTransfers(serviceDeps, account.id, knex)
      ).resolves.toEqual({
        credits: [depostTransfer],
        debits: []
      })
```
`depostTransfer` would need to be stored from the `beforeEach`

Reply:
Did something a little different without needing to store `depositTransfer`